src/metabase/models/permissions_group_membership.clj

```diff
@@ -27,8 +27,13 @@
                {:status-code 400})))))
 
 (defn- check-not-last-admin []
-  (when (<= (db/count PermissionsGroupMembership
-              :group_id (:id (perms-group/admin)))
+  (when (<= (:count
+             (first
+              (db/query {:select [:%count.*]
+                         :from   [[:permissions_group_membership :pgm]]
+                         :join   [[:core_user :user] [:= :user.id :pgm.user_id]]
+                         :where  [:and [:= :pgm.group_id (u/the-id (perms-group/admin))]
+                                       [:= :user.is_active true]]})))
```

Comment:
All this nonsense is not strictly part of the ticket, but I noticed that the existing check ensuring you couldn't remove the last admin (from the permissions group side, not the user API side) didn't care if the last admin was archived. This didn't seem right, so I changed it, but it'd be great to get confirmation from @flamber or someone.

Reply:
Yes, I have two expectations as noted in "Expected behavior" of the issue, which should prevent any problems; cannot deactivate yourself and make sure there's at least 1 active admin.